docs-client/src/lib/transports/transport.ts

```diff
@@ -14,19 +14,22 @@
  * under the License.
  */
 
-import { providers } from '../header-provider';
+import { docServiceDebug, providers } from '../header-provider';
+
 import { Method } from '../specification';
 
 export default abstract class Transport {
   public async send(
     method: Method,
     bodyJson: string,
     headers: { [name: string]: string },
+    endpointPath?: string,
+    queries?: string,
   ): Promise<string> {
     const providedHeaders = await Promise.all(
       providers.map((provider) => provider()),
     );
-    let filledHeaders = {};
+    let filledHeaders = { [docServiceDebug]: 'true' };
```

Comment:
Got it.!

Reply:
Could you elaborate this a little bit more please?
I used the `docServiceDebug` header to branch in the method `proxyToApi` in serve.ts when I sent a debug request. (before that we used the HTTP method which sends the request if it's POST.)
But how does the debug request contain the `WEBPACK_SERVE` information and we can use it to branch?

Reply:
Ah I just meant here, this can be `proxy.env.WEBPACK_SERVE === 'true' ? { [ Debug ] } : {};`

That variable should only be set when running the dev server so the production build won't have the debug header included (env checks are resolved at compile time by webpack).

Reply:
Ah~! I misunderstood. Thank you for the explanation~!

Reply:
It seems like the `process.env.WEBPACK_SERVE` does not propagate to the proxy server. Is there any way that I can hand on?

Reply:
Oh shoot I thought it was automatic - can you try adding this plugin to webpack.config.ts?

https://webpack.js.org/plugins/environment-plugin/

Reply:
@anuraaga It works~! Thank you for saving my time. :)